cashLetter.go

```diff
@@ -75,19 +76,31 @@ func (cl *CashLetter) setRecordType() {
 func (cl *CashLetter) Validate() error {
 	if cl.CashLetterHeader.RecordTypeIndicator == "N" {
 		if cl.GetBundles() != nil {
-			msg := fmt.Sprintf(msgCashLetterBundleEntries, cl.CashLetterHeader.RecordTypeIndicator)
-			return &CashLetterError{CashLetterID: cl.CashLetterHeader.CashLetterID,
-				FieldName: "RecordTypeIndicator", Msg: msg}
+			return &CashLetterError{
+				CashLetterID: cl.CashLetterHeader.CashLetterID,
+				FieldName:    "RecordTypeIndicator",
+				Msg:          fmt.Sprintf(msgCashLetterBundleEntries, cl.CashLetterHeader.RecordTypeIndicator),
+			}
 		}
 	}
 	switch cl.CashLetterHeader.CollectionTypeIndicator {
 	case
 		"00", "01", "02":
 	default:
 		if cl.GetRoutingNumberSummary() != nil {
-			msg := fmt.Sprintf(msgCashLetterRoutingNumber, cl.CashLetterHeader.CollectionTypeIndicator)
-			return &CashLetterError{CashLetterID: cl.CashLetterHeader.CashLetterID,
-				FieldName: "CollectionTypeIndicator", Msg: msg}
+			return &CashLetterError{
+				CashLetterID: cl.CashLetterHeader.CashLetterID,
+				FieldName:    "CollectionTypeIndicator",
+				Msg:          fmt.Sprintf(msgCashLetterRoutingNumber, cl.CashLetterHeader.CollectionTypeIndicator),
+			}
+		}
+	}
+
+	if cl.CashLetterControl == nil {
+		return &CashLetterError{
+			CashLetterID: cl.CashLetterHeader.CashLetterID,
+			FieldName:    "CashLetterControl",
+			Msg:          msgMandatoryRecord,
 		}
 	}
```

Comment:
Is there anything to validate within the CashLetterControl?

Reply:
There was an existing `Validate()` method for this record, so I just pushed an update to call it here.